pyk/src/pyk/ktool/kprove.py

```diff
@@ -138,19 +140,13 @@ def prove(
         command_env = os.environ.copy()
         command_env['KORE_EXEC_OPTS'] = kore_exec_opts
 
-        proc_output: str
-        try:
-            proc_output = run_process(command, _LOGGER, env=command_env).stdout
-        except CalledProcessError as err:
-            if err.returncode != 1:
-                raise RuntimeError(f'Command kprove exited with code {err.returncode} for: {spec_file}', err.stdout, err.stderr)
-            proc_output = err.stdout
+        proc_result = run_process(command, logger=_LOGGER, env=command_env, check=False, profile=self._profile)
```

Comment:
We're not ignoring all errors. We're always propogating stderr, and it will immediately fail on json decoding if it was any error code except `1` (which means "prover exited normally and produced output"). Any other error causes json decoding to fail (several lines down).

We have to not do json decoding here because we need to handle `dry_run` case.

Reply:
Ah! OK, great.